gateway/gateway.go

```diff
@@ -348,79 +358,158 @@ func (g *gateway) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 			cfg.Path = resolvedPath
 		}
 		path = cfg.Path
-		comType = lorahw.USB
+		comType = usb
 	}
+
+	args := []string{
+		fmt.Sprintf("--comType=%d", comType),
+		"--path=" + path,
+		fmt.Sprintf("--region=%d", region),
+	}
+
+	// find cgo exe
+	dir, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("error getting current directory: %w", err)
+	}
+	cgoexe := filepath.Join(dir, "cgo")
+
+	_, err = os.Stat(cgoexe)
+	if err != nil {
+		return fmt.Errorf("error getting the cgo exe: %w", err)
+	}
+
+	logReader, logWriter := io.Pipe()
+	bufferedLogReader := bufio.NewReader(logReader)
+	g.pipeReader = logReader
+	g.pipeWriter = logWriter
+	g.logReader = bufferedLogReader
+
+	config := pexec.ProcessConfig{
+		ID:        g.Name().Name,
+		Name:      cgoexe,
+		Args:      args,
+		Log:       false,
+		LogWriter: logWriter,
+	}
+
 	var comTypeString string
 	switch comType {
-	case lorahw.SPI:
+	case spi:
 		comTypeString = "SPI"
-	case lorahw.USB:
+	case usb:
 		comTypeString = "USB"
 	}
 
 	g.logger.Infof("starting %s gateway connected to path %s", comTypeString, path)
 
-	if err := lorahw.SetupGateway(comType, path, region); err != nil {
-		return fmt.Errorf("failed to set up the gateway: %w", err)
+	process := pexec.NewManagedProcess(config, g.logger)
+
+	if err := process.Start(ctx); err != nil {
+		return fmt.Errorf("failed to start binary: %w", err)
+	}
+
+	// This call blocks until the process prints the start up logs
+	port, err := g.watchLogs(ctx)
+	if err != nil {
+		return err
+	}
+
+	g.logger.Debugf("connecting to port: %s\n", port)
+	target := "localhost:" + port
+	conn, err := grpc.NewClient(target, grpc.WithTransportCredentials(insecure.NewCredentials()))
```

Comment:
Def not for this pr, but as a future change we should look into linux sockets. It’s what we use for modules but idk how to use them yet